k9mail/src/main/java/com/fsck/k9/activity/RecipientView.java

```diff
@@ -0,0 +1,296 @@
+        return ccWrapper.getVisibility() == View.VISIBLE;
+    }
+
+    public boolean isBccVisible() {
+        return bccWrapper.getVisibility() == View.VISIBLE;
+    }
+
+    public void showNoRecipientsError() {
+        toView.setError(toView.getContext().getString(R.string.message_compose_error_no_recipients));
+    }
+
+    public List<Address> getToAddresses() {
+        return Arrays.asList(toView.getAddresses());
+    }
+
+    public List<Address> getCcAddresses() {
+        return Arrays.asList(ccView.getAddresses());
+    }
+
+    public List<Address> getBccAddresses() {
+        return Arrays.asList(bccView.getAddresses());
+    }
+
+    public List<Recipient> getToRecipients() {
+        return toView.getObjects();
+    }
+    public List<Recipient> getCcRecipients() {
+        return ccView.getObjects();
+    }
+    public List<Recipient> getBccRecipients() {
+        return bccView.getObjects();
+    }
+
+    public void hideCryptoStatus() {
+        if (cryptoStatusView.getVisibility() == View.GONE) {
+            return;
+        }
+
+        cryptoStatusView.animate().translationX(100).setDuration(300).setListener(new AnimatorListenerAdapter() {
+            @Override
+            public void onAnimationEnd(Animator animation) {
+                super.onAnimationEnd(animation);
+                cryptoStatusView.setVisibility(View.GONE);
+            }
+        }).start();
+    }
+
+    public enum CryptoStatusType {
+        DISABLED(4), SIGN_ONLY(3), OPPORTUNISTIC_NOKEY(0), OPPORTUNISTIC_UNTRUSTED(1), OPPORTUNISTIC_TRUSTED(2);
+        final int childToDisplay;
+        CryptoStatusType(int childToDisplay) {
+            this.childToDisplay = childToDisplay;
+        }
+    }
+    public void showCryptoStatus(final CryptoStatusType childToDisplay) {
+        if (cryptoStatusView.getVisibility() == View.VISIBLE) {
+            switchCryptoStatus(childToDisplay);
+            return;
+        }
+
+        cryptoStatusView.setTranslationX(100);
+        cryptoStatusView.setVisibility(View.VISIBLE);
+        cryptoStatusView.animate().translationX(0).setDuration(300).setListener(new AnimatorListenerAdapter() {
+            @Override
+            public void onAnimationEnd(Animator animation) {
+                switchCryptoStatus(childToDisplay);
+            }
+        }).start();
+    }
+
+    private void switchCryptoStatus(CryptoStatusType cryptoStatus) {
+        int childToDisplay = cryptoStatus.childToDisplay;
+        if (cryptoStatusView.getDisplayedChild() != childToDisplay) {
+            cryptoStatusView.setDisplayedChild(childToDisplay);
+        }
+    }
+
+    /**
+     * Does the device actually have a Contacts application suitable for
+     * picking a contact. As hard as it is to believe, some vendors ship
+     * without it.
+     *
+     * @return True, if the device supports picking contacts. False, otherwise.
+     */
+    public boolean hasContactPicker() {
+        return activity.hasContactPicker();
```

Comment:
As far as I can tell there's no need for `hasContactPicker()` to live in the activity. Move it to the presenter and you can remove this method.